File: node/updateUser.js
const db = require('./utils/dbClient');

exports.handler = async (event) => {
  const { userId } = event.pathParameters;
  const { name, email } = JSON.parse(event.body);

  const result = await db.update({
    TableName: 'Users',
    Key: { userId },
    UpdateExpression: 'set #name = :name, email = :email',
    ExpressionAttributeNames: { '#name': 'name' },
    ExpressionAttributeValues: {
      ':name': name,
      ':email': email
    },
    ReturnValues: 'ALL_NEW'
  }).promise();

  return {
    statusCode: 200,
    body: JSON.stringify(result.Attributes),
  };
};

File: python/update_user.py
import json
import boto3
from boto3.dynamodb.conditions import Key

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('Users')

def lambda_handler(event, context):
    user_id = event['pathParameters']['userId']
    data = json.loads(event['body'])
    name = data.get('name')
    email = data.get('email')

    update_expr = []
    expr_attrs = {}

    if name:
        update_expr.append("name = :name")
        expr_attrs[':name'] = name

    if email:
        update_expr.append("email = :email")
        expr_attrs[':email'] = email

    if not update_expr:
        return {"statusCode": 400, "body": json.dumps({"message": "Nothing to update"})}

    response = table.update_item(
        Key={'userId': user_id},
        UpdateExpression="SET " + ", ".join(update_expr),
        ExpressionAttributeValues=expr_attrs,
        ReturnValues="ALL_NEW"
    )

    return {
        "statusCode": 200,
        "body": json.dumps(response['Attributes'])
    }
